### PR TITLE
Issue #26: Fixing enum remapping

### PR DESCRIPTION
**Pull request by brianjquinn**
This change is in reference to: https://github.com/tlivings/graphql-component/issues/26

Let me know if you need more explanation and/or a more adequate test.

I can add Changelog entries before merging as well.

### CODE REVIEW

File: lib/resolvers.js [hunk 1]
```diff
@@ -64,16 +64,18 @@ const wrapResolvers = function (bind, resolvers = {}) {
         wrapped[name] = {};
       }
 
-      for (const [resolverName, func] of Object.entries(value)) {
+      for (const [resolverName, funcOrPrimitive] of Object.entries(value)) {
```
Comment (tlivings) [inline lib/resolvers.js]: I don't think it is necessary to rename the variable.

File: lib/resolvers.js [hunk 2]
```diff
@@ -64,16 +64,18 @@ const wrapResolvers = function (bind, resolvers = {}) {
...
         if (['Query', 'Mutation', 'Subscription'].indexOf(name) > -1) {
           debug(`memoized ${name}.${resolverName}`);
-          wrapped[name][resolverName] = memoize(name, resolverName, func.bind(bind));
+          wrapped[name][resolverName] = memoize(name, resolverName, funcOrPrimitive.bind(bind));
           continue;
         }
-        wrapped[name][resolverName] = func.bind(bind);
+        // bind if the value mapped to a resolverName is a function
+        // otherwise dont to support internal enum value remapping
+        wrapped[name][resolverName] = funcOrPrimitive instanceof Function ? funcOrPrimitive.bind(bind) : funcOrPrimitive;
```
Comment (tlivings) [inline lib/resolvers.js]: Use `typeof func === 'function'` instead

Comment (tlivings) [reply lib/resolvers.js]: Specifically because you are checking for a function primitive not a function object. Use `typeof` to check primitives.

Comment (brianjquinn) [reply lib/resolvers.js]: Makes sense. Wrapping my head around when to use what (`typeof` vs `instanceof`) seems like it's suggested that you use `typeof` for built in types and `instanceof` for custom types? I'll make the change

File: lib/resolvers.js (hunk 1 above)
Comment (brianjquinn) [reply lib/resolvers.js]: no prob. I'll change it back.

**Pull request merged**